### ISSUE TITLE
Removal of living European data from project: Spatio-temporal immune zonation of the human kidney

### ISSUE THREAD

**Issue by morrisonnorman** [HumanCellAtlas/dcp#536]
# Issue
It has been brought to our attention that the [Spatio-temporal immune zonation of the human kidney] (https://data.humancellatlas.org/explore/projects/abe1a013-af7a-45ed-8c26-f3793c24a1f4) project contains data from living European donors. ALL instances of data associated with living European donors needs to be removed/deleted from DCP components.

# High Priority Task List
Remove all instances of data associated with living European donors from DCP components.
Deleted from:
- [ ] Upload
- [ ] Ingest
- [ ] Data Store
- [ ] Analysis
- [ ] Matrix Service
- [ ] Browser

A more detailed list of tasks and nascent Project Deletion SOP is being developed in [this google doc](https://docs.google.com/document/d/1Q2BYJJ8To_BOPA1tVksNHwnWFMxfFD3Yb1nwVbjnI-0/edit#)

# Requirements
The old project UUID should be maintained as this is already being referenced by 3rd parties. The project can be reingested once the living European data has been removed. The reingested data needs to be linked* to the the old project UUID.

#AUDR
AUDR capability does not currently handle partial deletion of data from projects.
Project deletion involves removing the entire project dataset and associated bundles.
Reingest and manually assign the existing project UUID to the newly ingested project.

# Next steps
Once the high priority deletion task list has been completed, the project can be reingested - with the living European donor metadata and data removed. Note that the reingested data needs to be linked* to the the old project UUID.

*MVP for reingestion is a direct replacement of the project UUID with no versioning.

**Comment by morrisonnorman** [comment]
@jahilton All of the living donors are European.

**Comment by jahilton** [comment]
Here's our living donor bundle list
```
499b690e-feee-4675-8531-882a62b6020c
fff16441-5212-4319-bf52-9e23d0eff7eb
5b8fc043-b00e-4699-b75f-19be0c7d9dc0
ce1f6515-6143-4e77-b162-fe6060cc10e6
4882ec6f-fe81-43d8-9446-595fbf6319da
27a43436-6e63-44ce-a2b3-80618f00b7f6
0d7aee1d-a3d7-4532-903a-c531ac07998d
9ca7449a-4b91-4cdd-b226-a7fbb7b1ffd0
8b7635a4-88a4-4c76-9946-6e3bebd51bfa
1a26883e-d034-491c-8d0a-0e93e854eacb
daaf63cf-04ae-4335-868c-32e63af0c37a
6f01e3ae-9743-4424-983b-18b18f0ffeed
002fe390-ba3f-4aa7-ae43-aa7949b09bb1
a1e2d3d2-3b47-4d20-a331-548818f3604b
caf3a1b6-60d3-42fd-a56c-48fe0e6de571
79ec8f41-fccc-46ed-af43-e63892238d77
a0a78a4e-7891-4ef2-a22d-1bfab43ffe47
6086e0e1-c679-44ad-bc62-57875b216fc4
7b4fd692-c38d-43c6-bb5b-bf3dcbc837eb
f269045b-1b3b-40d6-86ae-398a5b699e1d
c0e82caf-6fed-4524-9267-d97ac0a4d2f2
f6d920a5-63ab-4d4b-8fb7-0fe2fdbf2fcd
67cdc6a8-9f47-4dd9-908d-9654e489361d
97735720-4c1f-4b69-9463-e6c9febc2764
f838b9fe-3e03-41cf-8e1f-1f17db456975
3acc603d-b617-49ae-aa0d-353db0aa9e7e
880a3226-d432-4ab7-b81a-4c00421c6590
32b6880e-6c5c-41e4-b5b1-96b097cc52da
c99ea041-0daf-4b25-bee9-d9691af11a5e
5bf86ddb-a57b-49be-b734-8b09d7259e60
7ef3700c-d2d3-4e6e-b73e-c22f3380e7c3
70372f87-4cd4-40d7-8c09-1ebc641e1384
d8f7f78d-1c77-49cc-b934-28d0da11c6d0
8388fed2-1af2-4102-a578-82d3ef5ace03
9d993f96-67d2-4640-8217-89b1ecc60028
6963e95a-f81d-4bfc-9639-46da2a958d7a
f840f06a-58b1-4699-a672-ef0b08fb3304
8f585ec2-54c2-4d7b-8cea-0a9ceada6c8f
88aed2c9-6537-40fe-a04c-cd58900e9abf
a182995c-83c9-4da9-8029-f73b1d742d1b
00005cce-dfbe-40a9-922a-e1da11a11ada
ea25386a-dc17-453d-9312-b61995f2c3f6
328a2bf0-3f3c-4441-96a8-ddf41852e650
363aeb9a-010d-49ec-b8e0-9062552b9062
3c3061a7-1dc5-422e-bc14-4760b47f6ad2
81c5a0ed-50d3-4c60-b1b4-57df9862b636
62893d1d-91ea-4842-8d76-d6d4e67a795e
0dce2a49-1d81-45a2-98c3-75c1e154e96e
9e91edba-447e-47d3-8095-4406520e2fe9
9bd51e8f-abb2-40e6-8ee2-e0afa8b6caa6
```

**Comment by ESapenaVentura** [comment]
We don't have explicit data about the donor ethnicity, although it's safe to assume that all of them are European. I can look at the paper for further confirmation if necessary

**Comment by jahilton** [comment]
@ESapenaVentura if there's any uncertainty, can you contact Benjamin?

**Comment by morrisonnorman** [comment]
RE European donors: The samples were all collected from British Hospitals,
so they are all residents in Europe and subject to GDPR. All data associated data needs to be removed.

@jahilton Although he priority is to delete the 'living bundles', we don't have a mechanism to  delete specific bundles and update the metadata for the project.

I think it will have to be a 'delete all' from the project... and reingest.

**Comment by ESapenaVentura** [comment]
The context in this case is to delete bundles that contain the aforementioned donors, but the rest of the bundles (in case we can AUDR and not re-ingest) will be unaffected by this. Even if they remain unaffected, would you need new versions of them? @diekhans

**Comment by lauraclarke** [comment]
@ESapenaVentura do the project records specifically reference the cancer biopsys or link to the donors which will be deleted in a manner that isn't fixable with simple updates?

**Comment by ESapenaVentura** [comment]
I don't think I understand the question. Do you mean if the project metadata references the cancer biopsies?

**Comment by lauraclarke** [comment]
Yes, will the project level metadata be made inaccurate or any links in the deceased donor bundles be made invalid when the cancer biopsy bundles are deleted.

Those would be potential things which might require re-ingestion rather than deleting the bundles which reference the alive donors

If no deceased donor bundles reference living donors and any project-level metadata that references the cancer biopsy donors can be updated via simple updates then we might not need re-ingestion

**Comment by lauraclarke** [comment]
Okay based on discussions in slack and in person with Tony and Alegria there are other reasons that just deleting the needed bundles might lead to challenges later so I suspect this isn't a suitable solution

**Comment by jahilton** [comment]
Tickets have been created & linked to this Epic for
Ingest
DataStore
Browser
MatrixService
SecondaryAnalysis

@parthshahva does not believe Upload maintains any record so action may not be required. He will comment here once confirmed.

**Comment by parthshahva** [comment]
Per discussion in #data-wrangling @ESapenaVentura will be cleaning up the corresponding staging area in the wrangler workspace. I've confirmed that the production staging area was cleaned out after the submission/exporting completed and data was copied over to DSS.

**Comment by jlzamanian** [comment]
Wrangler issue ticket 
https://github.com/HumanCellAtlas/hca-data-wrangling/issues/344

**Comment by jlzamanian** [comment]
Wrangler issue ticket 
https://github.com/HumanCellAtlas/hca-data-wrangling/issues/344

**Comment by jlzamanian** [comment]
More comprehensive [Project Deletion SOP](https://docs.google.com/document/d/1NImUUZenaP6GWGkMaKLpzMzNAb1k5jLgo4IT7tgVLms/edit).
